### PR TITLE
Add Event Log

### PR DESCRIPTION
**Pull request by RyanUlch**
Added event log component with ability to add log line(s) from anywhere. As well as, having certain lines only be viewable on dev builds.

Unit Tests included.

### CODE REVIEW

File: src/components/EventLog/EventLog.vue [hunk 1]
```diff
@@ -0,0 +1,64 @@
+<!-- Update log is the screen the Player can see that provides context to what they are doing in the game -->
+<!-- Also provides ability to show dev logs, to help verify other components are functioning while building them -->
+
+<script setup lang="ts">
+import { computed } from 'vue';
```
Comment (RyanUlch) [inline src/components/EventLog/EventLog.vue]: Why?

File: src/composables/logComposable.ts [hunk 2]
```diff
@@ -1,25 +1,25 @@
-import { ref } from 'vue';
+import { reactive, ref } from 'vue';
+
+export type logLine = {
```
Comment (RyanUlch) [inline src/composables/logComposable.ts]: Capitalize first letter type

File: src/composables/logComposable.ts [hunk 3]
```diff
@@ -1,25 +1,25 @@
...
 // Which line to start log (after the log is cleared - log isn't deleted, just shows lines after clearing)
 const readFromLine = ref(0);
 
 // Composable handling UpdateLog
 export const useLogComposable = () => {
-	const addLogLine = (line: string) => {
-		log.value.push(line);
+	const addLogLine = (line: string, dev: boolean = false) => log.unshift({ line: line, dev: dev });
+	const addLogLines = (lines: logLine[]) => {
+		lines.map((logLine) => log.unshift({ line: logLine.line, dev: logLine.dev ? logLine.dev : false }));
```
Comment (RyanUlch) [inline src/composables/logComposable.ts]: Use forEach - no need for return value

Comment (RyanUlch) [reply src/composables/logComposable.ts]: Supply {} to forEach to not return value.

File: src/components/EventLog/EventLog.vue [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+	import { useLogComposable } from '../../composables/logComposable';
+	const { log, readFromLine } = useLogComposable();
+	const lines = computed(() => {
```
Comment (RyanUlch) [inline src/components/EventLog/EventLog.vue]: Could move computed() into composable

File: src/components/EventLog/EventLog.vue [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+		return log.filter((lineObj) => lineObj.dev === false);
+	});
+</script>
+
+<template>
+	<div class="outerContainer">
+		<div class="innerContainer">
+			<p
+				v-for="(logLine, lineIndex) in lines.slice(readFromLine, 100)"
+				:key="`line-${lineIndex}`"
```
Comment (RyanUlch) [inline src/components/EventLog/EventLog.vue]: remove 'line-' from key

File: src/components/EventLog/EventLog.vue [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+		background-color: black;
+		color: white;
+		border-radius: 1rem;
+		overflow: hidden;
+		border: 5px solid grey;
+	}
+	.innerContainer {
+		display: flex;
+		flex-direction: column-reverse;
+		align-items: flex-start;
```
Comment (RyanUlch) [inline src/components/EventLog/EventLog.vue]: Remove useless property

File: src/components/EventLog/EventLog.vue [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+		height: 300px;
```
Comment (RyanUlch) [inline src/components/EventLog/EventLog.vue:35]: Add Padding-right to inner container

**Review by RyanUlch — COMMENTED** [review]
Needs issues to be address

**Review by ericsartor — APPROVED** [review]

**Pull request merged**